package-builder/flox-build.mk

```diff
@@ -377,6 +394,76 @@ define BUILD_nix_sandbox_template =
 
 endef
 
+# The following template renders targets for the sandbox build mode.
+define BUILD_nix_expression_template =
+
+  $(eval _pvarname = $(subst -,_,$(_pname)))
+
+ 	$(eval _result = result-$(_pname))
+  $(eval $(_pvarname)_tmpBasename = $(TMPDIR)/$($(_pvarname)_shortHash)-$(_pname))
+  $(eval $(_pvarname)_buildMetaJSON = $($(_pvarname)_tmpBasename)-build-meta.json)
+
+  .PHONY: $(_pvarname)_nix_sandbox_build
+  $(_pvarname)_nix_expression_build: $($(_pvarname)_buildScript) $($(_pvarname)_src_tar) \
+		$(if $(_do_buildCache),$($(_pvarname)_buildCache))
+	@echo "Building $(_pname) in Nix expression (pure) mode"
+
+	$(_V_) set -o pipefail && \
+	$(_nix) build -L --file $(__nef)                 \
+	  --argstr nixpkgs-url "$(BUILDTIME_NIXPKGS_URL)" \
+	  --argstr system $(NIX_SYSTEM)                   \
+	  --argstr pkgs-dir $(NIX_EXPRESSION_DIR)         \
+	  --out-link "result-$(_pname)"                       \
+	  --json                                              \
+	  pkgs.$(_pname)                                      \
+	  |                                                   \
+	  $(_jq)                                              \
+			--argjson eval_meta                               \
+			  "$$$$(set -o pipefail &&                 \
+			  $(_nix) eval -L --file $(__nef)                 \
+			  --argstr nixpkgs-url "$(BUILDTIME_NIXPKGS_URL)" \
+			  --argstr system $(NIX_SYSTEM)                   \
+			  --argstr pkgs-dir $(NIX_EXPRESSION_DIR)         \
+			  --json                                          \
+			  --apply 'pkg: {                                 \
+			    version = pkg.version or "unknown";           \
+			    name = pkg.name;                              \
+			    pname = pkg.pname or pkg.name;                \
+			    meta = pkg.meta or null;                      \
+			  }'                                              \
+			  pkgs.$(_pname)                                  \
+			)" \
+	    '.[0] * { log:"/oops/no/logs" } * $$$$eval_meta' > $($(_pvarname)_buildMetaJSON)
```

Comment:
Embedded invocations like this are better done by splitting out into another target. Will refactor that by way of a separate commit.

Reply:
Split this out in commit fbb9ed791055531000fc6f0a939893f376a725e5